### ISSUE TITLE
Query: we should propagate optional navigations thru nav rewrite, if the initial optional navigation was created using SelectMany-GroupJoin-DefaultIfEmpty

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFramework#7975]
Repro:

```cs
    class Program
    {
        static void Main(string[] args)
        {
            using (var ctx = new MyContext())
            {
                ctx.Database.EnsureDeleted();
                ctx.Database.EnsureCreated();

                var users3 = (from user in ctx.Users
                              join ou in ctx.OrganisationUsers on user.Id equals ou.UserId into link
                              from organisation in link.DefaultIfEmpty()
                              select new
                              {
                                  Id = user.Id,
                                  Organisation = organisation == null ? "N/A" : organisation.Organisation.Name
                              }).ToList();
            }
        }
    }

    public class MyContext : DbContext
    {
        public DbSet<User> Users { get; set; }
        public DbSet<Organisation> Organisations { get; set; }
        public DbSet<OrganisationUser> OrganisationUsers { get; set; }

        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            modelBuilder.Entity<OrganisationUser>().HasKey(ou => new { ou.OrganisationId, ou.UserId });
            modelBuilder.Entity<OrganisationUser>().HasOne(ou => ou.Organisation).WithMany(o => o.OrganisationUsers).HasForeignKey(ou => ou.OrganisationId);
            modelBuilder.Entity<OrganisationUser>().HasOne(ou => ou.User).WithMany(u => u.OrganisationUsers).HasForeignKey(ou => ou.UserId);
        }

        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
           optionsBuilder.UseSqlServer(@"Server=.;Database=Repro7975;Trusted_Connection=True;MultipleActiveResultSets=true");
        }
    }

    public class User
    {
        public int Id { get; set; }
        public List<OrganisationUser> OrganisationUsers { get; set; }
    }

    public class Organisation
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public List<OrganisationUser> OrganisationUsers { get; set; }
    }

    public class OrganisationUser
    {
        public int OrganisationId { get; set; }
        public Organisation Organisation { get; set; }

        public int UserId { get; set; }
        public User User { get; set; }
    }
```

This produces the following sql:

```sql
SELECT [user].[Id], CASE
    WHEN [ou].[OrganisationId] IS NULL
    THEN N'N/A' ELSE [organisation.Organisation].[Name]
END
FROM [Users] AS [user]
LEFT JOIN [OrganisationUsers] AS [ou] ON [user].[Id] = [ou].[UserId]
INNER JOIN [Organisations] AS [organisation.Organisation] ON [ou].[OrganisationId] = [organisation.Organisation].[Id]
```

Instead we should detect that organization is optional created via SelectMany-GroupJoin-DefaultIfEmpty and create LEFT JOIN instead

**Comment by maumar** [comment]
Found when investigating #7830

**Comment by maumar** [comment]
fixed in bb7eebde422e6bf9601f4c97c72492dd52f0fe41
